clippy_lints/src/lib.rs

```diff
@@ -2069,6 +2072,8 @@ pub fn register_plugins(store: &mut rustc_lint::LintStore, sess: &Session, conf:
     store.register_late_pass(|| box unused_async::UnusedAsync);
     let disallowed_types = conf.disallowed_types.iter().cloned().collect::<FxHashSet<_>>();
     store.register_late_pass(move || box disallowed_type::DisallowedType::new(&disallowed_types));
+    let import_renames = conf.import_renames.iter().cloned().collect::<FxHashSet<_>>();
+    store.register_late_pass(move || box import_rename::ImportRename::new(&import_renames));
```

Comment:
```suggestion
    let import_renames = conf.import_renames.clone();
    store.register_late_pass(move || box import_rename::ImportRename::new(import_renames));
```
No need to transform here or in the constructor. Pass by value so you don't need to clone again either.

Reply:
Annoyingly you can't because of `pass: impl Fn() -> LateLintPassObject + 'static + sync::Send + sync::Sync,`. You have to clone it then move the ref in. I wonder if it would be worth `Box::leak(conf)` then at least the few lints that need non `Copy` stuff could just take a ref and only clone once?

Reply:
Oh I see. The function has to be able to be invoked multiple times. Maybe do this.
```suggestion
    let import_renames = conf.import_renames.clone(); // might not need this clone
    store.register_late_pass(move || box import_rename::ImportRename::new(import_renames.clone()));
```